waddrmgr/db.go

```diff
@@ -623,39 +711,49 @@ func deserializeImportedAddress(addressID []byte, row *dbAddressRow) (*dbImporte
 		return nil, managerError(ErrDatabase, str, nil)
 	}
 
-	retRow := dbImportedAddressRow{
+	retRow := dbImportedPubKeyAddressRow{
 		dbAddressRow: *row,
 	}
 
-	pubLen := binary.LittleEndian.Uint32(row.rawData[0:4])
+	var offset uint32
+	pubLen := binary.LittleEndian.Uint32(row.rawData[offset : offset+4])
+	offset += 4
 	retRow.encryptedPubKey = make([]byte, pubLen)
-	copy(retRow.encryptedPubKey, row.rawData[4:4+pubLen])
-	offset := 4 + pubLen
+	copy(retRow.encryptedPubKey, row.rawData[offset:offset+pubLen])
+	offset += pubLen
+
 	privLen := binary.LittleEndian.Uint32(row.rawData[offset : offset+4])
 	offset += 4
 	retRow.encryptedPrivKey = make([]byte, privLen)
 	copy(retRow.encryptedPrivKey, row.rawData[offset:offset+privLen])
+	offset += privLen
 
 	return &retRow, nil
 }
 
-// serializeImportedAddress returns the serialization of the raw data field for
+// serializeImportedPubKeyAddress returns the serialization of the raw data field for
 // an imported address.
-func serializeImportedAddress(encryptedPubKey, encryptedPrivKey []byte) []byte {
+func serializeImportedPubKeyAddress(encryptedPKHash, encryptedPubKey, encryptedPrivKey []byte) []byte {
 	// The serialized imported address raw data format is:
 	//   <encpubkeylen><encpubkey><encprivkeylen><encprivkey>
 	//
-	// 4 bytes encrypted pubkey len + encrypted pubkey + 4 bytes encrypted
-	// privkey len + encrypted privkey
+	// 4 bytes encrypted pubkey len + encrypted pubkey +
+	// 4 bytes encrypted privkey len + encrypted privkey
 	pubLen := uint32(len(encryptedPubKey))
 	privLen := uint32(len(encryptedPrivKey))
 	rawData := make([]byte, 8+pubLen+privLen)
-	binary.LittleEndian.PutUint32(rawData[0:4], pubLen)
-	copy(rawData[4:4+pubLen], encryptedPubKey)
-	offset := 4 + pubLen
+
+	var offset uint32
```

Comment:
I would revert this code to the old way it calculated offset, to me it reads clearer.  Shorter and less state to consider.